client/components/Players/PlaybackInfoCard.vue

```diff
@@ -0,0 +1,417 @@
+                    sessionInfo.TranscodingInfo.TranscodeReasons.length
+                  )
+                "
+              />
+              <div class="d-flex flex-column">
+                <div
+                  v-for="(reason, index) in sessionInfo.TranscodingInfo
+                    .TranscodeReasons"
+                  :key="index"
+                >
+                  {{ $t(camelCase(reason)) }}
+                </div>
+              </div>
+            </div>
+            <div
+              v-if="
+                sessionInfo.TranscodingInfo &&
+                sessionInfo.TranscodingInfo.Framerate
+              "
+              class="d-flex"
+            >
+              <div
+                class="font-weight-bold mr-2"
+                v-text="$t('playbackInfo.transcodingFramerate.name')"
+              />
+              <div>
+                {{
+                  $t('playbackInfo.transcodingFramerate.value', {
+                    value: sessionInfo.TranscodingInfo.Framerate
+                  })
+                }}
+              </div>
+            </div>
+            <div
+              v-if="
+                sessionInfo.TranscodingInfo &&
+                sessionInfo.TranscodingInfo.CompletionPercentage
+              "
+              class="d-flex flex-column"
+            >
+              <div class="d-flex">
+                <div
+                  class="font-weight-bold mr-2"
+                  v-text="$t('playbackInfo.transcodingProgress')"
+                />
+                <div>
+                  {{
+                    sessionInfo.TranscodingInfo.CompletionPercentage.toFixed(
+                      1
+                    ) + '%'
+                  }}
+                </div>
+              </div>
+              <v-progress-linear
+                class="d-inline-block mt-2"
+                :value="
+                  sessionInfo.TranscodingInfo.CompletionPercentage.toFixed(1)
+                "
+              />
+            </div>
+          </v-card-text>
+        </v-card>
+      </v-col>
+    </v-row>
+  </v-container>
+</template>
+
+<script lang="ts">
+import { SessionInfo } from '@jellyfin/client-axios';
+import Vue from 'vue';
+import { mapGetters, mapState } from 'vuex';
+import { camelCase } from 'camel-case';
+
+export default Vue.extend({
+  data() {
+    return {
+      updateSessionInterval: null as number | null,
+      sessionInfo: null as SessionInfo | null,
+      playerStats: {},
+      videoDimensions: { width: 0, height: 0 }
+    };
+  },
+  computed: {
+    ...mapState('deviceProfile', ['deviceId']),
+    ...mapState('playbackManager', ['currentMediaSource']),
+    ...mapGetters('playbackManager', [
+      'getCurrentVideoTrack',
+      'getCurrentAudioTrack',
+      'getCurrentSubtitleTrack'
+    ]),
+    isTranscoding(): boolean {
+      return !!(
+        this.sessionInfo?.PlayState?.PlayMethod === 'Transcode' ||
+        this.sessionInfo?.TranscodingInfo
```

Comment:
Wouldn't the or and equality check already return a boolean?

Reply:
Apparently not, TypeScript was complaining about TranscodingInfo returning an object, hence the `!!` here :)